libr/anal/p/anal_x86_cs.c

```diff
@@ -76,31 +76,35 @@ static bool is_xmm_reg(cs_x86_op op) {
  * @param  n       Operand index
  * @param  set     if 1 it adds set (=) to the operand
  * @param  setoper Extra operation for the set (^, -, +, etc...)
- * @return         char* with the esil operand
+ * @param  out     Output
+ * @return         Void
  */
-static char *getarg(struct Getarg* gop, int n, int set, char *setop) {
+static void getarg(struct Getarg* gop, int n, int set, char *setop, char *out) {
 	char *setarg = setop ? setop : "";
 	cs_insn *insn = gop->insn;
 	csh handle = gop->handle;
 	char buf[64];
 	cs_x86_op op;
+	buf[0] = 0;
 
 	if (!insn->detail) {
-		return NULL;
+		out[0]=0;
```

Comment:
Maybe it's not best way to replace ``return NULL;`` ? Under normal conditions, this function can return an empty string? Like ``return strdup("");``